eip-0014.md

```diff
@@ -0,0 +1,540 @@
+
+#### Liquidity pool bootstrapping
+
+A liquidity pool is bootstrapped in two steps:
+
+1. Initial amounts of tokens are deposited to the pool bootstrapping proxy contract. For the initializing deposit the amount of LP tokens is calculated using special formula which is `LP = sqrt(X_deposited * Y_deposited)`. Also in order to track pro-rata LP shares of the total reserves of a new pair a unique token called "LP token" must be issued. As soon as tokens can’t be re-issued on Ergo the whole LP emission has to be done at once.
+2. Initial reserves of token pair and a remainder of LP tokens left after initial depositing are passed to the pool contract. Also pool NFT is issued. Thus the pool is created in an initial state. Correctness of the state is checked by the proxy contract (This is why it is important to check whether a pool was initialized through the proxy contract. Luckily it can be done easily using pool NFT ID).
+
+In order to avoid blowing up the pool contract with a code which handles only specific intialization aspects a dedicated type of contract is used.
+
+```
+                    InitialInput [X:N, Y:M]
+ 1.                             |
+                                |
+            PoolBootProxy [LP:1000000000000000000, X:N, Y:M]
+ 2.             |                                     |
+                |                                     |
+    LPRewardOut [LP:sqrt(N*M)]     Pool [NFT:1, LP:1000000000000000000-sqrt(N*M), X:N, Y:M]
+```
+
+#### Schema of the pool bootstrapping UTXO
+
+Section        | Description
+---------------|------------------------
+value          | Constant amount of ERGs
+tokens[0]      | LP token reserves
+tokens[1]      | X token initial reserves
+tokens[2]      | Y token initial reserves
+R5[Coll[Byte]] | Blake2b256 hash of the pool script
+R6[Long]       | Desired amount of shares LP expects to receive
+R7[Long]       | Pre-configured pool fee multiplier numerator
+R8[Long]       | Miner fee
+R9[Coll[Byte]] | Initiator proposition bytes
+ 
+#### Pool bootstrapping contract
+
+```scala
+{
+    val poolScriptHash  = SELF.R5[Coll[Byte]].get
+    val desiredSharesLP = SELF.R6[Long].get
+    val poolFeeConfig   = SELF.R7[Long].get
+    val minerFeeNErgs   = SELF.R8[Long].get
+    val initiatorProp   = SELF.R9[Coll[Byte]].get
+
+    val selfLP = SELF.tokens(0)
+    val selfX  = SELF.tokens(1)
+    val selfY  = SELF.tokens(2)
+
+    val tokenIdLP = selfLP._1
+
+    val pool           = OUTPUTS(0)
+    val sharesRewardLP = OUTPUTS(1)
+
+    val maybePoolLP  = pool.tokens(1)
+    val poolAmountLP =
+        if (maybePoolLP._1 == tokenIdLP) maybePoolLP._2
+        else 0L
+
+    val validPoolContract  = blake2b256(pool.propositionBytes) == poolScriptHash
+    val validPoolErgAmount = pool.value == SELF.value - sharesRewardLP.value - minerFeeNErgs
+    val validPoolNFT       = pool.tokens(0) == (SELF.id, 1L)
```

Comment:
Why is there need to generate the pool NFT while spending the bootstrapping box? Can we not have the pool NFT generated beforehand and stored in the bootstrap box as `self.tokens(3)`? 
That way, everyone can verify **before** spending this box that the pool NFT was indeed generated in quantity 1, which is much more secure.